Notation.Plugin.AzureKeyVault/KeyVault/KeyVaultClient.cs

```diff
@@ -0,0 +1,110 @@
+        // Certificate client (lazy initialization)
+        private Lazy<CertificateClient> _certificateClient;
+        // Cryptography client (lazy initialization)
+        private Lazy<CryptographyClient> _cryptoClient;
+
+        private const string invalidInputErrorMessage = "Invalid input. The valid input format is '{\"contractVersion\":\"1.0\",\"keyId\":\"https://<vaultname>.vault.azure.net/<keys|certificate>/<name>/<version>\"}'";
+
+        /// <summary>
+        /// Constructor to create AzureKeyVault object from key identifier or
+        /// certificate identifier.
+        ///
+        /// <param name="id">
+        /// Key identifier or certificate identifier. (e.g. https://<vaultname>.vault.azure.net/keys/<name>/<version>)
+        /// </param>
+        /// </summary>
+        public KeyVaultClient(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException(nameof(id), "Id must not be null or empty");
+            }
+
+            var uri = new Uri(id);
+            // Validate uri
+            if (uri.Segments.Length != 4)
+            {
+                throw new ValidationException(invalidInputErrorMessage);
+            }
+            if (uri.Segments[1] != "keys/" && uri.Segments[1] != "certificates/")
+            {
+                throw new ValidationException(invalidInputErrorMessage);
+            }
+            if (uri.Scheme != "https")
+            {
+                throw new ValidationException(invalidInputErrorMessage);
+            }
+
+            // Extract keys|certificates name from the uri
+            this._name = uri.Segments[2].TrimEnd('/');
+            this._version = uri.Segments[3].TrimEnd('/');
+            var keyVaultUrl = $"{uri.Scheme}://{uri.Host}";
+            this._keyId = $"{keyVaultUrl}/keys/{_name}/{_version}";
+
+            // initialize credential and lazy clients
+            var credential = new DefaultAzureCredential();
+            this._certificateClient = new Lazy<CertificateClient>(() => new CertificateClient(new Uri(keyVaultUrl), credential));
+            this._cryptoClient = new Lazy<CryptographyClient>(() => new CryptographyClient(new Uri(_keyId), credential));
+        }
+
+        /// <summary>
+        /// Constructor to create AzureKeyVault object from keyVaultUrl, name 
+        /// and version.
+        /// </summary>
+        public KeyVaultClient(string keyVaultUrl, string name, string version) : this($"{keyVaultUrl}/keys/{name}/{version}") { }
+
+        /// <summary>
+        /// Sign the payload and return the signature.
+        /// </summary>
+        public async Task<byte[]> Sign(SignatureAlgorithm algorithm, byte[] payload)
+        {
+            var signResult = await _cryptoClient.Value.SignDataAsync(algorithm, payload);
+
+            if (signResult.KeyId != _keyId)
+            {
+                throw new PluginException("Invalid keys or certificates identifier.");
```

Comment:
Do we need a `.` suffix in the exception message?

Reply:
Removed `.`